### PR TITLE
Do not enforce memory limits for fault tolerant queries

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

LowMemoryKiller is responsible for freeing up the memory if necessary

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Core

> How would you describe this change to a non-technical end user or system administrator?

`-`

## Related issues, pull requests, and links

`-`

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java [hunk 1]
```diff
@@ -394,12 +397,20 @@ private TaskInfo doUpdateTask(
         SqlTask sqlTask = tasks.getUnchecked(taskId);
         QueryContext queryContext = sqlTask.getQueryContext();
         if (!queryContext.isMemoryLimitsInitialized()) {
-            long sessionQueryMaxMemoryPerNode = getQueryMaxMemoryPerNode(session).toBytes();
+            RetryPolicy retryPolicy = getRetryPolicy(session);
+            if (retryPolicy == TASK) {
+                // Memory limit for fault tolerant queries is enforced by the MemoryPool.
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java:402]: What do you mean by "Memory limit for fault tolerant queries is enforced by the MemoryPool"?

Comment (arhimondr) [reply core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java:402]: I meant "Memory limit for fault tolerant queries should be enforced only by the MemoryPool". Let me rephrase it.

**Review by losipiuk — COMMENTED** [review]
Should we change some documentation?

**Comment by arhimondr** [comment]
Updated

**Review by losipiuk — APPROVED** [review]
